apps/web-giddh/src/app/new-inventory/component/recipe/create-recipe/create-recipe.component.html

```diff
@@ -78,124 +78,266 @@ <h1 class="section-head pb-2 w-100">{{ localeData?.finished_stock }}</h1>
+                                            >{{ commonLocaleData?.app_by_product_output }}
+                                        </mat-panel-title>
+                                    </mat-expansion-panel-header>
+                                    <div class="consumption-details-wrapper w-100">
+                                        <div class="mt-2">
+                                            <table class="table-raw-stock">
+                                                <thead>
+                                                    <tr>
+                                                        <th>{{ commonLocaleData?.app_name }}</th>
+                                                        <th>{{ commonLocaleData?.app_variant }}</th>
+                                                        <th>{{ commonLocaleData?.app_quantity }}</th>
+                                                        <th>{{ commonLocaleData?.app_unit }}</th>
+                                                        <th></th>
+                                                    </tr>
+                                                </thead>
+                                                <tbody>
+                                                    <tr
+                                                        *ngFor="
+                                                            let byProductLinkedStock of recipe.byProducts;
+                                                            let k = index
                                                         "
-                                                        *ngIf="recipe.linkedStocks?.length > 1"
-                                                        (click)="removeLinkedStock(i, j)"
                                                     >
-                                                        <i class="icon-trash"></i>
-                                                    </a>
-                                                </td>
-                                            </tr>
-                                        </tbody>
-                                    </table>
-                                    <div>
-                                        <a href="javascript:void(0);" (click)="addNewLinkedStockInRecipe(recipe)">
-                                            {{ localeData?.add_stock }}
-                                        </a>
+                                                        <td>
+                                                            <div>
+                                                                <select-field
+                                                                    [cssClass]="'form-control mat-field-border'"
+                                                                    [name]="'stock_' + i + '_' + k"
+                                                                    [scrollableElementId]="'stock_' + i + '_' + k"
```

Comment:
name and scrollableElementId is similar to linked stocks for stock and variant etc...